src/details/ArborX_MinimumSpanningTree.hpp

```diff
@@ -585,7 +585,14 @@ struct MinimumSpanningTree
   {
     Kokkos::Profiling::pushRegion("ArborX::MST::MST");
 
-    BVH<MemorySpace> bvh(space, primitives);
+#if KOKKOS_VERSION >= 30700
+    using Access = AccessTraits<Primitives, PrimitivesTag>;
+    constexpr int dim = GeometryTraits::dimension<
+        typename Details::AccessTraitsHelper<Access>::type>::value;
+    using Box = ExperimentalHyperGeometry::Box<dim>;
```

Comment:
Now that I see how many times you have that code I almost want to define a `Box<DIM>` for pre 3.7 Kokkos

Reply:
It would help some, but would still have some clutter as long as we have to maintain backwards compatibility with older geometries. Hopefully, that's for a short time until Kokkos 4.0 release.

Reply:
```C++
namespace ExperimentalHyperGeometry {
template <int DIM>
struct Box : struct ArborX::Box {
  using AborX::Box::Box;
};
}

Reply:
I'm not sure what you mean. We can't inherit as we need to use new point types as corners.

Reply:
in the Kokkos pre-3.7 code path

Reply:
Ah, OK. With a static assert `DIM == 3`? And I assume same with Point and Sphere. I can do that, but would you mind if I do that after we merge this PR?

Reply:
PR was rebased on #734.